src/Psalm/Type/Atomic/TKeyedArray.php

```diff
@@ -45,27 +45,19 @@ class TKeyedArray extends Atomic
      */
     public $class_strings;
 
-    /**
-     * True if the objectlike has been created from an explicit array,
-     * or if we're sure that this objectlike has only these keys.
-     *
-     * @var bool
-     */
-    public $sealed = false;
```

Comment:
Can't we keep the attribute and maintain invariant `$this->sealed = $this->fallback_key_type === null && $this->fallback_value_type === null;` instead? Shouldn't be too much of a hassle with an immutable object, I imagine.

I find it much more readable to check `if ($type->sealed) ...` instead of `if ($type->fallback_value_type === null) ...`.

Reply:
Or replace it with a method like this:
```php
public function isSealed(): bool {
    return $this->fallback_key_type === null && $this->fallback_value_type === null;
}
```

Reply:
I'll replace `$this->fallback_key_type` and `$this->fallback_value_type` with `$this->fallback_params`, which will at least make it a slightly prettier check (and it'll match up with my Hack equivalent data structure)